mpas_analysis/shared/generalized_reader/generalized_reader.py

```diff
@@ -124,9 +126,31 @@ def open_multifile_dataset(fileNames, calendar, simulationStartTime=None,
                                  startDate=startDate,
                                  endDate=endDate)
 
-    ds = xarray.open_mfdataset(fileNames,
-                               preprocess=preprocess_partial,
-                               decode_times=False, concat_dim='Time', **kwargs)
+    kwargs = {'decode_times': False,
+              'concat_dim': 'Time'}
+
+    softLimit = resource.getrlimit(resource.RLIMIT_NOFILE)[0]
+
+    # use autoclose if we will use more than half the soft limit on open files
+    autoclose = len(fileNames) > softLimit/2
```

Comment:
@milenaveneziani, I think you should just think of the "soft limit" as being the limit on the number of files that can be opened.  It's technically possible to change this from what I have read but that doesn't mean it's a good idea.  There's a "hard" limit that you can't exceed (unless you're root).  I don't know what our options would be to change the "soft" limit on the HPC machines.  I think we should assume it's a hard limit for us.

I added a comment clarifying this line.